From 5822243fdb2ffd0f8ef7d773716a8338542fbfd3 Mon Sep 17 00:00:00 2001
From: YangYonggang <b31664@freescale.com>
Date: Fri, 9 Mar 2012 14:52:10 +0800
Subject: [PATCH 251/635] ENGR00176478 The name of back camera and front
 camera was switched.

The root cause was the name info was got before switching sensor.
It should be got after switching action.

Signed-off-by: YangYonggang <b31664@freescale.com>
---
 mx6/libcamera/V4l2CsiDevice.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/mx6/libcamera/V4l2CsiDevice.cpp b/mx6/libcamera/V4l2CsiDevice.cpp
index 16696e6..587dce8 100755
--- a/mx6/libcamera/V4l2CsiDevice.cpp
+++ b/mx6/libcamera/V4l2CsiDevice.cpp
@@ -83,6 +83,16 @@ namespace android{
                     if ((fd = open(dev_node, O_RDWR, O_NONBLOCK)) < 0)
                         continue;
                     CAMERA_HAL_LOG_RUNTIME("dev_node is %s", dev_node);
+
+                    if (fd > 0){
+                        mCameraDevice = fd;
+                        ret = V4l2SetSensor(cameraId);
+                    }
+                    else{
+                        CAMERA_HAL_ERR("The device name is not correct or the device is error");
+                        return CAPTURE_DEVICE_ERR_OPEN;
+                    }
+
                     if(ioctl(fd, VIDIOC_DBG_G_CHIP_IDENT, &vid_chip) < 0 ) {
                         close(fd);
                         fd = 0;
-- 
1.8.0

